serializers/src/main/java/io/pravega/schemaregistry/codec/Codecs.java

```diff
@@ -103,14 +101,13 @@ public CodecType getCodecType() {
         }
 
         @Override
-        public ByteBuffer encode(ByteBuffer data) throws IOException {
+        public void encode(ByteBuffer data, ByteArrayOutputStream bos) throws IOException {
             int capacity = Snappy.maxCompressedLength(data.remaining());
-            ByteBuffer encoded = ByteBuffer.allocate(capacity);
+            byte[] encoded = new byte[capacity];
             
             int size = Snappy.compress(data.array(), data.arrayOffset() + data.position(),
```

Comment:
Can we add a precondition check that size <= capacity?

Reply:
capacity is a non negative number returned by snappy library. we dont need to do a precondition.

Reply:
yeah, checked the code of Snappy and it too invokes Snappy.maxCompressedLength internally.